scripts/bench_runner.py

```diff
@@ -0,0 +1,1111 @@
+                        Quantity(f"{b[metric]}{unit}")
+                    )
+
+    # Populate metrics from the baseline runs.
+    for run in base_runs:
+        for b in run["benchmarks"]:
+            name = b["name"]
+            # Baseline runs don't have comparable benchmarks, so we only need
+            # to populate the 'base' list for main benchmarks.
+            if name in benchmark_names:
+                for metric in metrics:
+                    unit = b.get("time_unit", "") if "time" in metric else ""
+                    benchmark_metrics[metric][name].base.append(
+                        Quantity(f"{b[metric]}{unit}")
+                    )
+
+    return benchmark_metrics
+
+
+def print_metric_key(
+    console: Console,
+    alpha: float,
+    has_baseline: bool,
+    comp_mapping: ComparableBenchmarkMapping,
+) -> None:
+    """Prints a legend for the metrics table.
+
+    This explains the format of the output table, including what the delta,
+    median, and confidence interval values represent.
+
+    Args:
+        console: The rich console to print to.
+        alpha: The alpha value for statistical significance.
+        has_baseline: Whether a baseline benchmark was run.
+    """
+    console.print("Metric key:")
+
+    conf = int(100 * (1.0 - alpha))
+
+    name = "BenchmarkName..."
+    delta_icon = str(DeltaKind.IMPROVEMENT)
+    delta = "[faster]<delta>[/faster]"
+    p = "p=<U-test P-value>"
+    base_median = "[base_median]<median>[/base_median]"
+    base_conf = f"[base_conf]<% at {conf}th conf>[/base_conf]"
+    exp_median = "[exp_median]<median>[/exp_median]"
+    exp_conf = f"[exp_conf]<% at {conf}th conf>[/exp_conf]"
+
+    key_table = Table.grid(
+        Column(justify="right"),
+        Column(),
+        Column(),
+        Column(),
+        Column(),
+        padding=(0, 1),
+    )
+    if has_baseline:
+        key_table.add_row(name, delta_icon, delta, "", p)
+        key_table.add_row("baseline:", "", base_median, "±", base_conf)
+        key_table.add_row("experiment:", "", exp_median, "±", exp_conf)
+    else:
+        key_table.add_row(name, "", exp_median, "±", exp_conf)
+        # Only display comparable key if we have comparables to display.
+        if bool(comp_mapping.name_to_comp_tag):
+            key_table.add_row("vs Comparable:", delta_icon, delta, p)
+            key_table.add_row("", "", base_median, "±", base_conf)
+    console.print(Padding(key_table, (0, 0, 1, 3)))
+
+
+def print_results_table(
+    console: Console,
+    alpha: float,
+    has_baseline: bool,
+    metrics: list[str],
+    benchmark_names: list[str],
+    benchmark_metrics: dict[str, dict[str, BenchmarkRunMetrics]],
+    comp_mapping: ComparableBenchmarkMapping,
+) -> None:
+    """Builds and prints the main results table.
+
+    This function constructs a rich `Table` to display the benchmark results,
+    including deltas, medians, and confidence intervals for each metric. It thin
```

Comment:
Did you mean `then`?

```suggestion
    including deltas, medians, and confidence intervals for each metric. It then
```